springfox-core/src/main/java/springfox/documentation/schema/Model.java

```diff
@@ -47,7 +48,8 @@ public Model(
       String baseModel,
       String discriminator,
       List<String> subTypes,
-      String example) {
+      String example,
+      Model parent) {
```

Comment:
I'd like tho use `ModelRef` here directly assuming some other plugin is responsible to dig out the `Model` from the `@ApiModel#parent()`. Typically by modifying the `ModelDependencyProvider` i.e. Parent => is a dependency of => Subclass

Secondly, we're anyways turning around and resolving it to a RefModel in the mapper.